packages/css/src/index.ts

```diff
@@ -215,20 +223,23 @@ const transforms = [
   {}
 )
 
-const responsive = (styles: Exclude<ThemeUIStyleObject, UseThemeFunction>) => (
-  theme?: Theme
-) => {
-  const next: Exclude<ThemeUIStyleObject, UseThemeFunction> = {}
+const responsive = (
+  styles: Exclude<ThemeUIStyleObject, ThemeDerivedStyles>
+) => (theme?: Theme) => {
+  const next: Exclude<ThemeUIStyleObject, ThemeDerivedStyles> = {}
   const breakpoints =
     (theme && (theme.breakpoints as string[])) || defaultBreakpoints
   const mediaQueries = [
     null,
-    ...breakpoints.map(n => `@media screen and (min-width: ${n})`),
+    ...breakpoints.map((n) => `@media screen and (min-width: ${n})`),
   ]
 
-  for (const key in styles) {
-    const value =
-      typeof styles[key] === 'function' ? styles[key](theme) : styles[key]
+  for (const k in styles) {
+    const key = k as keyof typeof styles
```

Comment:
A type of key in X isn't always assignable to keys of type of X, even if we forget about the prototype. This part of TypeScript is a bit inconvenient, but it does help with some bugs.

Explanation by Anders Hejlsberg:
https://github.com/microsoft/TypeScript/pull/12253#issuecomment-263132208